compiler/rustc_abi/src/lib.rs

```diff
@@ -1241,6 +1241,31 @@ impl Abi {
     pub fn is_scalar(&self) -> bool {
         matches!(*self, Abi::Scalar(_))
     }
+
+    /// Returns the fixed alignment of this ABI, if any
+    pub fn inherent_align<C: HasDataLayout>(&self, cx: &C) -> Option<AbiAndPrefAlign> {
+        Some(match *self {
+            Abi::Scalar(s) => s.align(cx),
+            Abi::ScalarPair(s1, s2) => {
+                AbiAndPrefAlign::new(cmp::max(s1.align(cx).abi, s2.align(cx).abi))
```

Comment:
```suggestion
                s1.align(cx).max(s2.align(cx))
```
(you don't want to disturb the preferred alignment - to test the effect this you'll sadly need a type with different abi-vs-pref alignment, like `u64` on a 32-bit target, so not the easiest thing)